### PR TITLE
Relax depdency constraints

### PR DESCRIPTION
**Pull request by danyx23**
In library type projects, version numbers should only be restricted to a fixed major version number if there is a good reason to do so. This commit relaxes the constraints and turns fixed major version numbers (^1.3.4) into lower bounds (>=1.3.4)

### CODE REVIEW

**Review by Marigold — APPROVED** [review]
The only question here is whether to pin pandas to a specific version here (since owid-datautils-py uses it a lot, especially for operations with nans that can be sensitive to pandas upgrades). Up to @pabloarosado, I think it's fine to keep it unpinned.

**Comment by danyx23** [comment]
@Marigold I think the best way to do it is to pin the version at the "application" point whatever that is - so if you use e.g. a notebook in the notebooks repo then there you should maybe pin pandas to 1.4 or whatever and this library will happily comply (since 1.4 is in the valid range).

If we were to pin it here then if you want to bump it in one app to 1.5 then all other users of the library would be forced to upgrade as well. I'm cool with waiting until the usage sites had time to pin their pandas version but I'm against pinning it on the library level even for this library.

**Comment by Marigold** [comment]
@danyx23 good argument, agree.

**Comment by pabloarosado** [comment]
I would need to review whether all of our functions do what they are supposed to, given the recent issues with pandas and nans. We could consider adding more tests specifically for behaviour related to nans, which is the most dangerous one and version-sensitive. But I'm happy to merge this as-is to unblock the whole thing.

**Review by pabloarosado — APPROVED** [review]
Looks good.

**Pull request merged**
